platform/engineering/design-docs/2020-08-11-redirects-from-cms.md

```diff
@@ -0,0 +1,224 @@
+        }
+    ],
+    "metatag": {
+        "value": {
+            "title": "Edit URL redirect | Veterans Affairs",
+            "twitter_cards_type": "summary_large_image",
+            "og_site_name": "Veterans Affairs",
+            "twitter_cards_title": "Edit URL redirect | Veterans Affairs",
+            "twitter_cards_site": "@DeptVetAffairs",
+            "og_title": "Edit URL redirect | Veterans Affairs"
+        }
+    }
+}
+
+```
+
+The only modification needed at the CMS export is to add a destination element that would take drupal node paths
+```
+ "redirect_redirect": [
+        {
+            "uri": "internal:\/node\/281",
+            "title": "Server",
+            "options": []
+         }
+```
+
+and translate `"uri": "internal:\/node\/281",` to an actual path like
+```
+"destination": [
+        {
+            "uri": "pittsburgh-health-care\/about\/volunteer-or-donate",
+         }
+```
+This would be a trivial to add.
+
+#### II. Vets-website content creation.
+  _(Covered in High Level Design section.)_
+
+#### III. Vets-website deploy to AWS
+  _(Covered in High Level Design section.)_
+
+
+### Code Location
+
+Step I: Addition of relative path for destination would be added to the [CMS repo here](https://github.com/department-of-veterans-affairs/va.gov-cms/blob/master/docroot/modules/custom/va_gov_content_export/src/TomeExporter.php#L97)
+
+Step II: Build the HTML files with meta refresh during content file generation in the [vets-website repo here](https://github.com/kevwalsh/vets-website/blob/1d6a53cd9b53ec71b69ae2bce78bd5694eaf1d7f/src/site/stages/build/drupal/metalsmith-drupal.js#L51).
+
+Step III:  Building the AWS S3 objects with redirect metadata would happen in the [Devops repo here](https://github.com/department-of-veterans-affairs/devops/blob/a58dfbc762466905d5727cb4cd626d5e6dad9c3f/ansible/deployment/roles/deploy-vets-website/files/vets-website-deploy.sh)
+which is called from [here](https://github.com/department-of-veterans-affairs/devops/blob/6fd9fc8f5ebcb2b685e4e0a20450c472655fcef2/ansible/deployment/roles/deploy-vets-website/tasks/main.yml#L2-L7)
+
+
+### Testing Plan
+The current content release method already does link checking.  This method would be tested to a large degree by this existing test.  The CMS already does a significant amount of testing of a front end build in a non-AWS environment so we would add a Behat test to makes sure the that the client side redirect is working.
+
+Additional testing could be added on the Vets-website deployment tests to verify that a set of redirects work properly on the server side redirects in AWS.
+
+### Logging
+Logging would not be required.  There may be some statistics that could be tracked within AWS, but that may not be necessary.  Google Analytics may also be used to track instances of displaying the 404 page that may in some cases include referral data.  Server logs track HTTP requests and are likely already used to track requests ending in 404 errors.
+
```

Comment:
This may not be related to logging per se, but we need to be able to manage the redirects over time as they age out, so we aren't managing thousands of redirects over time.

Reply:
Any redirects known to drupal can be manged in a nice searchable interface
![image](https://user-images.githubusercontent.com/5752113/91583768-a897cc80-e91f-11ea-80e1-98218b9818af.png)